core/unit_test/TestHalfConversion.hpp

```diff
@@ -3,6 +3,9 @@
 
 #ifndef TESTHALFCONVERSION_HPP_
 #define TESTHALFCONVERSION_HPP_
+
+#include <impl/Kokkos_Half_FloatingPointWrapper.hpp>
```

Comment:
This header defines macros that are used here. We should consider moving them to `Kokkos_Macros.hpp`.